src/types/mod.rs

```diff
@@ -113,48 +89,52 @@ impl Symbol for Op {
 #[derive(Debug, Clone, Eq)]
 pub struct Var {
     id: DeBruijn,
-    name: Name_,
+    name: Option<String>,
 }
 impl Var {
-    pub fn new(id: DeBruijn, name: Name_) -> Self {
+    pub fn new(id: DeBruijn, name: Option<String>) -> Self {
         Var { id, name }
     }
+    pub fn named(mut self, name: String) -> Self {
+        self.name = Some(name);
+        self
+    }
 }
 impl Hash for Var {
     fn hash<H: Hasher>(&self, state: &mut H) {
         self.id.hash(state);
     }
 }
-impl Name for Var {
-    fn name(&self) -> Name_ {
-        self.name.clone()
-    }
-}
```

Comment:
Same as with `Op` - why take away the `name`?

Reply:
(this is back in my code review commit — it moved to the body of `impl Variable`)